tools/spectral/ipa/__tests__/createMethodShouldNotHaveQueryParameters.test.js

```diff
@@ -0,0 +1,146 @@
+import testRule from './__helpers__/testRule';
+import { DiagnosticSeverity } from '@stoplight/types';
+
+const componentSchemas = {
+  schemas: {
+    Schema: {
+      type: 'object',
+    },
+  },
+  parameters: {
+    QueryParam: {
+      name: 'query-param',
+      in: 'query',
+      schema: {
+        type: 'string',
+      },
+    },
+    PathParam: {
+      name: 'resource-id',
+      in: 'path',
+      schema: {
+        type: 'string',
+      },
+    },
+  },
+};
+
+testRule('xgen-IPA-106-create-method-should-not-have-query-parameters', [
+  {
+    name: 'valid methods',
+    document: {
+      components: componentSchemas,
+      paths: {
+        '/resource': {
+          post: {
+            parameters: [
+              {
+                name: 'header-param',
+                in: 'header',
+                schema: { type: 'string' },
+              },
+              {
+                name: 'resource-id',
+                in: 'path',
+                schema: {
+                  $ref: '#/components/schemas/Schema',
+                },
+              },
+              {
+                $ref: '#/components/parameters/PathParam',
+              },
+            ],
+          },
+        },
+        '/resource2': {
+          post: {
+            parameters: [],
+          },
+        },
+      },
+    },
+    errors: [],
+  },
+  {
+    name: 'invalid methods',
+    document: {
+      components: componentSchemas,
+      paths: {
+        '/resource': {
+          post: {
+            parameters: [
+              {
+                name: 'filter',
+                in: 'query',
+                schema: { type: 'string' },
+              },
+            ],
+          },
+        },
+        '/resource2': {
+          post: {
+            parameters: [
+              {
+                name: 'header-param',
+                in: 'header',
+                schema: { type: 'string' },
+              },
+              {
+                $ref: '#/components/parameters/QueryParam',
```

Comment:
It appears that your example is missing definition of #/components/parameters/QueryParam.
Assuming QueryParam from name might not be valid use case and we need to ensure that referenced parameter is in fact query parameter. Let me know if you want me to generate example for you and correct this unit test.